### DISCUSSION TITLE
"x files downloaded" but nowhere to be found

### DISCUSSION THREAD

**Discussion by JHMikkel** [danielfernau/unifi-protect-video-downloader#147] [General]
Complete newbie here!
I finally got the script working through docker and it seems to be running perfectly and doing what it's supposed to be doing. Except the files I am supposed to be getting is nowhere to be found.

This is the script I run:
`docker run --volume /Users/mikkel/Desktop/uni unifitoolbox/protect-archiver download --address=10.16.18.23 --username="username" --password="password" --cameras=60366473001c9303e70xxxxx --start=2022-04-13T07:00:00 --end=2022-04-13T07:01:00 --touch-files /Users/mikkel/Desktop/uni`

And this is the response I get:
```
Getting camera list
Successfully retrieved data from https://10.16.18.23:443/proxy/protect/api/cameras
Cameras found:
- Parkering (60366473001c9303e70xxxxx)
Downloading video files between 2022-04-13 07:00:00 and 2022-04-13 07:01:00 from 'https://10.16.18.23:443/proxy/protect/api/video/export' for camera Parkering
Downloading footage for camera 'Parkering' (60366473001c9303e70xxxxx)
Created path /Users/mikkel/Desktop/uni/2022/04/13/Parkering (03f1)
Downloading video for time range 2022-04-13 07:00:00 - 2022-04-13 07:00:59 to /Users/mikkel/Desktop/uni/2022/04/13/Parkering (03f1)/Parkering (03f1) - 2022-04-13 - 07.00.00.mp4
Download successful after 0s (7.0 mb, 10.0 mbps)
1 files downloaded (7.0 mb), 0 files skipped, 0 files failed, 1 files total
```

So all seems fine, but the destination folder is completely empty.

What am I doing wrong?

I am on macOS 12.2.1 running Docker Desktop 4.7.1.

PS. If/when this works, it's an amazing tool and definately something that should be native to Protect. The current way to export video is ridiculous.

**Comment by danielfernau** [comment]
Hi there!

Your command looks good so far, except for the volume mount and the path at the end of the command. The path at the end always needs to be `/downloads`, since this is the path to where the tools downloads the files _inside_ the Docker container. And with the `--volume` parameter you tell Docker where to map the `/downloads` path of the container to on your local filesystem.

Your command downloads the files to `/Users/mikkel/Desktop/uni` _inside_ the Docker container, that's why you can't find them. They are there, just not in a place where you can easily access them.

The command needs to look like this
`docker run --volume /Users/mikkel/Desktop/uni:/downloads unifitoolbox/protect-archiver download --address=10.16.18.23 --username="username" --password="password" --cameras=60366473001c9303e70xxxxx --start=2022-04-13T07:00:00 --end=2022-04-13T07:01:00 --touch-files /downloads`

The `--volume /Users/mikkel/Desktop/uni:/downloads` part tells Docker to mount a `/downloads` folder inside the container and map it to `/Users/mikkel/Desktop/uni` on your host machine, i.e. your macOS. And `/downloads` at the very end of the command tells my tool to write all files to this mounted file share.

Hope this helps; let me know if you need any additional assistance!

Thanks a lot -- and yes, I couldn't agree more. This should be a native feature. They could ship it as an extra desktop app in addition to the browser UI and include a bunch of useful features. But that's unlikely to happen, sadly. I mean, how useful are all the recordings in the end if you can't easily mass-export and/or archive them as needed... :)

**Reply by JHMikkel** [reply]
That's it... it works! 🥳

Once again, thank you!

And yeah, I think you're right. It is unlikely to happen.
But this is a really useful tool, and one I'm sure people would gladly spend money to use.

As far as I can tell, there isn't anything in this tool that Ubiquity would frown upon? I mean, what is holding you back from turning this into a full-fledgedd app (along with GUI)?

Oh, and thank you for the quick response - the turtle-tagging on your profile does not do you justice 😉

**Reply by danielfernau** [reply]
Perfect, glad to hear! You're welcome, and thanks.

No, there's nothing they could be mad about, as far as I see it. It interacts with their API the same way the browser does, just with a bit more logic and automation all around. I've changed the primary namespace of the project a few months ago as to respect (and not over-use) their brand names and trademarks. Other than that, things should be fine.

The main thing that holds me back is lack of free time, actually. That's also the reason for the "slow to respond" tag on my profile :)
Also, there are many other great community-built scripts out there which interact with all the different UI products and make things easier to work with, so I thought I might as well join those and help out the community a bit. This tool will stay free and open-source for the foreseeable future, and everyone's welcome to fix and add things. 

Have a great weekend!

**Reply by JHMikkel** [reply]
Well I can definitely relate to the lack of free time.

But let me know if you ever think about turning this into an app/service/website. Paid or free, I would be willing to invest some hours :)

And then just a last follow up question..
It seems the script will only export recordings as long as the cameras are connected to the NVR. Is there any way around that?
I mean, the recordings must still be there even if the camera is connected or not?

**Reply by danielfernau** [reply]
Sure, will do. Thanks for the offer!

As far as I know, no. See also https://github.com/danielfernau/unifi-protect-video-downloader/issues/67#issuecomment-926063449
But I think it should somehow be possible to grab recordings for offline cameras. As long as you don't delete a camera, and it's just offline, the recordings are there (unless they are overwritten due to age or deleted based on retention settings). I'll definitely have another look at it since I think there could be some use-cases where such a feature would be helpful (broken or stolen camera, for example).
